### ISSUE TITLE
Resolved promise with error code 1

### ISSUE THREAD

**Issue by stefanjudis** [macbre/phantomas#569]
Hello,

one of the users found another bug in `grunt-phantomas` which leads to phantomas itself.
Case is, that when using the npm module we found another occurrence of a resolved promise, which then represents the status code 1.

https://github.com/stefanjudis/grunt-phantomas/issues/116

To give you an example.
```
var phantomas = require( 'phantomas' );

console.log( phantomas.version );

phantomas(
  'http://www.carfax.es/lp-coches-de-segunda-mano.html',
  {
    timeout: 30,
    'film-strip': true,
    'film-strip-dir': 'phantomas/images/1445626863104'
  }
).then(
  function() {
    console.log( 'Resolved' );
    console.log( arguments )
  },
  function() {
    console.log( 'Rejected' );
    console.log( arguments )
  }
);
```

leads to

```
stefan @ stefan-mac: ~/Sites/grunt-phantomas master (╯°□°)╯
> node test.js                                                                  [21:04:32]
1.12.0
Resolved
{ '0': { code: 1, json: undefined, results: undefined } }
```

Thanks. :)

**Comment by macbre** [comment]
@stefanjudis, can you check this issue using phantomas 1.13 (i.e. the latest version)?

**Comment by stefanjudis** [comment]
Sure - give me 10minutes. :)

**Comment by stefanjudis** [comment]
Unfortunately not. :(

Executed same file:

```
var phantomas = require( 'phantomas' );

console.log( phantomas.version );

phantomas(
  'http://www.carfax.es/lp-coches-de-segunda-mano.html',
  {
    timeout: 30,
    'film-strip': true,
    'film-strip-dir': 'phantomas/images/1445626863104'
  }
).then(
  function() {
    console.log( 'Resolved' );
    console.log( arguments )
  },
  function() {
    console.log( 'Rejected' );
    console.log( arguments )
  }
);
```

Console:

```
stefan @ stefan-mac: ~/Sites/grunt-phantomas master (╯°□°)╯
> node test.js                                                                            [21:21:54]
1.13.0
Resolved
{ '0': { code: 1, json: undefined, results: undefined } }
```

**Comment by macbre** [comment]
Interesting, can you run the script using `DEBUG=* node test.js` and provide a gist link with the full output?

**Comment by stefanjudis** [comment]
Sure. Here we go!

https://gist.github.com/stefanjudis/f588e2c1871e7db37841

**Comment by macbre** [comment]
It does work for me when I pass `engine: "webkit2"`:

```
  phantomas:ipc exit: [0,null] +64ms
  phantomas:results Process returned code #0 +8ms
  phantomas:results Resolving a promise with 0 exit code +28ms
```

I've added a small debugging improvement to "Error when parsing JSON" error in 408d8ab

**Comment by stefanjudis** [comment]
Am I right, that this will be released with phantomas `v1.14.0` then?

**Comment by macbre** [comment]
Correct and no problem :)
